node/src/accountant/big_int_db_processor.rs

```diff
@@ -0,0 +1,1613 @@
+        self.pure_rusqlite_params(
+            self.wei_change_params
+                .iter()
+                .map(|(name, value)| (name.as_str(), value as &dyn ToSql)),
+        )
+    }
+
+    fn pure_rusqlite_params_corrected(
+        &'a self,
+        wei_change_params: &'a [(&'a str, i64); 2],
+    ) -> impl Iterator<Item = (&str, &dyn ToSql)> {
+        self.pure_rusqlite_params(
+            wei_change_params
+                .iter()
+                .map(|(name, num)| (*name, num as &dyn ToSql)),
+        )
+    }
+
+    fn pure_rusqlite_params(
+        &'a self,
+        wei_change_params: impl Iterator<Item = (&'a str, &'a dyn ToSql)>,
+    ) -> impl Iterator<Item = (&'a str, &'a dyn ToSql)> {
+        self.params_except_wei_change
+            .iter()
+            .map(|(name, value)| (*name, value as &dyn ToSql))
+            .chain(wei_change_params)
+    }
+}
+
+pub trait DAOTableIdentifier: Debug + Send {
+    fn table_name() -> String;
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum WeiChange {
+    Addition(&'static str, u128),
+    Subtraction(&'static str, u128),
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct BigIntDbError(pub String);
+
+macro_rules! insert_update_error_from {
+    ($implementer: ident) => {
+        impl From<BigIntDbError> for $implementer {
+            fn from(iu_err: BigIntDbError) -> Self {
+                $implementer::RusqliteError(iu_err.0)
+            }
+        }
+    };
+}
+
+insert_update_error_from!(PayableDaoError);
+insert_update_error_from!(ReceivableDaoError);
+
+pub fn collect_and_sum_i128_values_from_table(
+    conn: &dyn ConnectionWrapper,
+    table: &str,
+    parameter_name: &str,
+) -> i128 {
+    let select_stm = format!(
+        "select {0}_high_b, {0}_low_b from {1}",
+        parameter_name, table
+    );
+    conn.prepare(&select_stm)
+        .expect("select stm error")
+        .query_map([], |row| {
+            Ok(BigIntDivider::reconstitute(
+                row.get::<usize, i64>(0).expectv("high bytes"),
+                row.get::<usize, i64>(1).expectv("low_bytes"),
+            ))
+        })
+        .expect("select query failed")
+        .flatten()
+        .sum()
+}
+
+macro_rules! create_big_int_sqlite_fns {
+    ($conn: expr, $flags: expr, $($sqlite_fn_name: expr),+; $($intern_fn_name: ident),+) => {
+        $($conn.create_scalar_function::<_, i64>($sqlite_fn_name, 2, $flags, move |ctx| {
+            Ok(BigIntDivider::$intern_fn_name(common_arg_distillation(
+                ctx,
+                $sqlite_fn_name,
+            )?))
+        })?;)+
+    }
+}
+
+macro_rules! parse_fn_creation_args {
+    ($ctx: expr, $fn_name: expr, $($idx: expr),+; $($parser: ident),+; $($err_msg: literal),+) => {
+        ($($ctx.get_raw($idx)
+            .$parser()
+            .map_err(|_| invalid_input_error($fn_name, format!($err_msg, $ctx.get_raw($idx))))?
+        ),+)
+    };
+}
+
+pub struct BigIntDivider {}
```

Comment:
Turns out there's an easier way. Whew!

Reply:
Too bold to state it, from the view of last findings I've got.